crates/sargon/src/profile/v100/entity/has_entity_security_state.rs

```diff
@@ -0,0 +1,24 @@
+use crate::prelude::*;
+
+pub trait HasEntitySecurityState {
+    fn security_state(&self) -> EntitySecurityState;
+
+    fn virtual_hierarchical_deterministic_factor_instances(
+        &self,
```

Comment:
add a `filter_kind: impl Into<Option<CAP26KeyKind>>`  and then instead of returning `factor_instances` you can:

```rust
let Some(key_kind) = filter_key_kind.into() else { return  factor_instances };
factor_instances.into_iter().filter(|fi| f.derivation_path().key_kind == key_kind).collect()
```